sway-error/src/error.rs

```diff
@@ -1732,6 +1744,81 @@ impl ToDiagnostic for CompileError {
                     format!("{}- references, direct or indirect, to arrays. E.g., `&[u64;3]` or `&&&[u64;3]`.", Indent::Single),
                 ],
             },
+            FieldAccessOnNonStruct { actually, storage_variable, field_name, span } => Diagnostic {
+                reason: Some(Reason::new(code(1), "Field access requires a struct".to_string())),
+                issue: Issue::error(
+                    source_engine,
+                    span.clone(),
+                    format!("{} has type \"{actually}\", which is not a struct{}.",
+                        if let Some(storage_variable) = storage_variable {
+                            format!("Storage variable \"{storage_variable}\"")
+                        } else {
+                            "This expression".to_string()
+                        },
+                        if storage_variable.is_some() {
+                            ""
+                        } else {
+                            " or a reference to a struct"
+                        }
+                    )
+                ),
+                hints: vec![
+                    Hint::info(
+                        source_engine,
+                        field_name.span(),
+                        format!("Field access happens here, on \"{field_name}\".")
+                    )
+                ],
+                help: if storage_variable.is_some() {
+                    vec![
+                        "Fields can be accessed only on storage variables that are structs.".to_string(),
+                    ]
+                } else {
+                    vec![
+                        "In Sway, fields can be accessed on:".to_string(),
+                        format!("{}- structs. E.g., `my_struct.field`.", Indent::Single),
+                        format!("{}- references, direct or indirect, to structs. E.g., `(&my_struct).field` or `(&&&my_struct).field`.", Indent::Single),
+                    ]
+                }
+            },
+            StorageFieldDoesNotExist { field_name, available_fields, storage_decl_span } => Diagnostic {
+                reason: Some(Reason::new(code(1), "Storage field does not exist".to_string())),
+                issue: Issue::error(
+                    source_engine,
+                    field_name.span(),
+                    format!("Storage field \"{field_name}\" does not exist in the storage.")
```

Comment:
It sounds a bit redundant indeed, but I put it there so that it follows the same patter like the message for structs. And also without it it sounded like ending abruptly. Definitely a matter of taste.